app/assets/javascripts/galleries/uploader.js

```diff
@@ -17,8 +18,9 @@ $(document).ready(function() {
     var usedUrls = $.map($('form.icon-upload').find('input[id$=_url]'), function(input) { return $(input).val(); });
     var uploadedUrls = $.map(uploadedIcons, function(value, key) { return key; });
     var unusedUrls = uploadedUrls.filter(function(x) { return usedUrls.indexOf(x) < 0; });
-    if (unusedUrls.length < 1) return true;
+    if (unusedUrls.length < 1) { return true; }
     deleteUnusedIcons($.map(unusedUrls, function(url) { return uploadedIcons[url]; }));
+    return false;
```

Comment:
I think `return false` changes behavior here so the submit doesn't finish: https://api.jquery.com/on/#event-handler suggests that this will call stopPropagation + preventDefault, whereas the default return (undefined) doesn't have that behavior - https://github.com/jquery/jquery/blob/ace646f6e83e653f666ba715c200739f1cbdba52/src/event.js#L326

Reply:
i... what... but... why...

Reply:
Not sure! jQuery seems to use undefined as equivalent to true in this situation, and return false is a different behavior.